week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,83 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
-
-/* 
-DO NOT EDIT ANYTHING ABOVE THIS LINE
-WRITE YOUR CODE BELOW
-*/
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
+//I tried to create new Array Method which covers both .filter() and .map() methods.
 
+Array.prototype.filterPlus = function(callback, propertyName) {
+  var output = [];
+  for (var element of this) {
+    if (callback(element)) {
+      output.push(element[propertyName]);
+    }
+  }
+  return output;
+};
 
```

Comment:
Wow! This is quite a solution. It works and is pretty advanced - top marks for trying something outside the box and getting your head around some new concepts. 👍

- Great use of comments to indicate what your thought process was and what you were trying to achieve. 
- I really like the way you thought about this problem. Particularly, that using `filter()` and then `map()` seemed inefficient and tried to do something where you didn't have to iterate over the same array twice.
- Well done - the use of `this`, callbacks and prototypes - there are plenty of advanced topics in there!!

Reply:
Thanks again for your inspiring words. I just wanted to try to implement the things I have learned about the Constructor functions and prototypes. If you didn`t comment like this I started to think that I made something useless. Thanks.